File: app/src/main/java/com/dirtydish/app/dirtydish/RecyclerAdapter.java
package com.dirtydish.app.dirtydish;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import android.widget.Toast;

import java.util.ArrayList;
import java.util.List;


class RecyclerViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener, View.OnLongClickListener{

    public TextView chore_name;

    private ItemClickListener itemClickListener;

    public RecyclerViewHolder(View itemView) {
        super(itemView);
        chore_name = (TextView)itemView.findViewById(R.id.chore_name);

        itemView.setOnClickListener(this);
        itemView.setOnLongClickListener(this);
    }

    public void setItemClickListener(ItemClickListener itemClickListener) {
        this.itemClickListener = itemClickListener;
    }

    @Override
    public void onClick(View v) {
        itemClickListener.onClick(v, getAdapterPosition(), false);
    }

    @Override
    public boolean onLongClick(View v) {
        itemClickListener.onClick(v, getAdapterPosition(), true);
        return true;
    }
}

public class RecyclerAdapter extends RecyclerView.Adapter<RecyclerViewHolder> {

    private List<Chore> listData;
    private Context context;

    public RecyclerAdapter(List<Chore> listData, Context context){
        this.listData = listData;
        this.context = context;
    }

    @NonNull
    @Override
    public RecyclerViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater inflater = LayoutInflater.from(parent.getContext());
        View itemView = inflater.inflate(R.layout.chore, parent, false);
        return new RecyclerViewHolder(itemView);
    }

    @Override
    public void onBindViewHolder(@NonNull RecyclerViewHolder holder, int position) {
        holder.chore_name.setText(listData.get(position).getName());

        holder.setItemClickListener(new ItemClickListener(){
            @Override
            public void onClick(View view, int position, boolean isLongClick) {
                if(isLongClick)
                    Toast.makeText(context, "Long Click" + listData.get(position).getName(), Toast.LENGTH_SHORT).show();
                else
                    Toast.makeText(context, "Short Click" + listData.get(position).getName(), Toast.LENGTH_SHORT).show();
            }
        });
    }

    @Override
    public int getItemCount() {
        return listData.size();
    }
}

File: app/src/main/java/com/dirtydish/app/dirtydish/RecyclerAdapter.kt
package com.dirtydish.app.dirtydish

import android.content.Context
import android.content.Intent
import android.support.v7.widget.RecyclerView
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import android.widget.Toast

private val tag = "RECYCLER_ADAPTER"

class RecyclerViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView), View.OnClickListener, View.OnLongClickListener {

    var chore_name: TextView
    var frequency: TextView
    //val choreTimeFrame: TextView
    val choreAssignee: TextView

    private var itemClickListener: ItemClickListener? = null

    init {
        chore_name = itemView.findViewById<View>(R.id.chore_name) as TextView
        frequency = itemView.findViewById<View>(R.id.chore_frequency) as TextView
        //choreTimeFrame = itemView.findViewById<View>(R.id.chore_time_frame) as TextView
        choreAssignee = itemView.findViewById<View>(R.id.chore_assignee) as TextView

        itemView.setOnClickListener(this)
        itemView.setOnLongClickListener(this)
    }

    fun setItemClickListener(itemClickListener: ItemClickListener) {
        this.itemClickListener = itemClickListener
    }

    override fun onClick(v: View) {
        itemClickListener!!.onClick(v, adapterPosition, false)
    }

    override fun onLongClick(v: View): Boolean {
        itemClickListener!!.onClick(v, adapterPosition, true)
        return true
    }
}

class RecyclerAdapter(private val listData: List<Chore>, private val context: Context) : RecyclerView.Adapter<RecyclerViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerViewHolder {
        val inflater = LayoutInflater.from(parent.context)
        val itemView = inflater.inflate(R.layout.chore_row, parent, false)
        return RecyclerViewHolder(itemView)
    }

    override fun onBindViewHolder(holder: RecyclerViewHolder, position: Int) {
        val currentChore = listData[position]
        holder.chore_name.text = currentChore.name
        var frequency_text = ""
        val freq = currentChore.frequency
        if (freq % 7 == 0) {
            frequency_text = (freq / 7).toString()
            if (freq == 7) frequency_text = "week"
            else frequency_text += " weeks"

        } else if (freq % 30 == 0) {
            frequency_text = (freq / 30).toString()
            if (freq == 30) frequency_text = "month"
            else frequency_text += " months"

        } else {
            frequency_text = freq.toString()
            if (freq == 1) frequency_text = "day"
            else frequency_text += " days"
        }

        holder.frequency.text = frequency_text

        if (currentChore.participants.size > 0) {
            //TODO: ACTUALLY ROTATE BETWEEN ALL CHORE PARTICIPANTS
            holder.choreAssignee.text = currentChore.participants[0].name
        } else {
            holder.choreAssignee.text = "none"
        }

        holder.setItemClickListener(object : ItemClickListener {
            override fun onClick(view: View, position: Int, isLongClick: Boolean) {
                Log.i("FREQUENCY", "frequency=" + currentChore.frequency)
                if (isLongClick) {
                    Toast.makeText(context, "Long Click" + currentChore.name, Toast.LENGTH_SHORT).show()
                    val intent = Intent(context.applicationContext,ModifyChore::class.java)
                            .putExtra("id", currentChore.id)
                            .putExtra("name", currentChore.name)
                            .putExtra("frequency", currentChore.frequency)
                            .putExtra("participants", currentChore.participants.toTypedArray())
                    Log.d(tag, context.toString())
                    context.startActivity(intent)
                } else {
                    Toast.makeText(context, "Short Click" + currentChore.name, Toast.LENGTH_SHORT).show()
                    val intent = Intent(context.applicationContext, ModifyChore::class.java)
                            .putExtra("id", currentChore.id)
                            .putExtra("name", currentChore.name)
                            .putExtra("frequency", currentChore.frequency)
                            .putExtra("participants", currentChore.participants.toTypedArray())
                    Log.d(tag, context.toString())
                    context.startActivity(intent)
                }
            }
        })
    }

    override fun getItemCount(): Int {
        return listData.size
    }

}
